### PR TITLE
Fixes call tiles overflow

### PR DESCRIPTION
**Pull request by luixxiul**
Fixes vector-im/element-web#20254

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

This PR fixes the issue that buttons and time string overflow the rectangle by making makes call tiles semi-responsive.

Note this achieves the responsiveness incompletely as the minimum width is fixed.

Please see below to check how strings which can be long according to translations are going to be treated.

**Image 1**: On bubble message layout
after1

**Image 2**: On modern layout
after2

**Image 3**
after3

(Button style inconsistency was addressed with another PR: https://github.com/matrix-org/matrix-react-sdk/pull/8129)

**Image 4 (extreme case)**

To demonstrate the elements would not overflow in an extreme case such as below:

after4

**Image 5 (IRC)**
after5

**Image 6 (with E2E icon on bubble message layout on LTR UI)**
after

https://user-images.githubusercontent.com/3362943/160240590-7cd173b3-6d67-4987-ae61-bf3bc0c881d6.mp4

type: defect

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fixes call tiles overflow ([\#8096](https://github.com/matrix-org/matrix-react-sdk/pull/8096)). Fixes vector-im/element-web#20254. Contributed by @luixxiul.

Preview: https://pr8096--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: res/css/views/messages/_CallEvent.scss [hunk 1]
```diff
@@ -19,22 +19,24 @@ limitations under the License.
...
         display: flex;
         flex-direction: row;
+        flex-wrap: wrap;
         align-items: center;
         justify-content: space-between;
+        gap: 4px 0;
 
+        position: relative;
+        margin: 4px 0;
+        padding: 12px 24px;
```
Comment (turt2live) [inline res/css/views/messages/_CallEvent.scss:31]: we should probably use our spacing variables now that we have them

**Review by turt2live — COMMENTED** [review]
generally seems fine, though I think we should be using our `$spacing` variables where possible

File: res/css/views/messages/_CallEvent.scss (hunk 1 above)
Comment (luixxiul) [reply res/css/views/messages/_CallEvent.scss:31]: I see. I am wondering what the policy on using and applying those variables is. Will they be applied gradually and incrementally to see how they should work, or at one time globally?

**Comment by turt2live** [comment]
(please avoid force pushing after review has been given, as it means we need to re-review the whole thing again)

**Comment by luixxiul** [comment]
There should be another way of doing this, but I have been not sure of another way of fixing the conflict... Sorry for disturbing!

**Review by turt2live — CHANGES_REQUESTED** [review]
This PR still isn't using the spacing variables throughout like we'd expect for new/modified code.

File: res/css/views/messages/_CallEvent.scss [hunk 2]
```diff
@@ -70,57 +79,62 @@ limitations under the License.
...
-            min-width: 0;
+            width: fit-content;
+            max-width: 100%;
 
             .mx_CallEvent_info_basic {
                 display: flex;
                 flex-direction: column;
+                gap: $spacing-4;
                 margin-left: 10px; // To match mx_CallEvent
+                margin-right: 10px;
```
Comment (luixxiul) [inline res/css/views/messages/_CallEvent.scss:117]: This is the same value as `margin-left: 10px;` just above.

**Review by turt2live — APPROVED** [review]
Thanks!

**Pull request merged**